### PR TITLE
implimented remote entries(folders are a work in progress)

### PR DESCRIPTION
**Pull request by Shalev001**

### CODE REVIEW

**Review by Fluff-E — APPROVED** [review]

File: FileProcessor/src/main/java/com/group7/FileProcessor/entries/RemFile.java [hunk 1]
```diff
@@ -4,40 +4,86 @@
...
+        updateContents();
+        this.entryId = entryid;
+
         this.repositoryId = repositoryId;
     }
 
-    public String getAccessKey() {
-        return accessKey;
+    public int getAccessKey() {
+        return entryId;
```
Comment (Tacitor) [inline FileProcessor/src/main/java/com/group7/FileProcessor/entries/RemFile.java:82]: Is this the right thing to return?

File: FileProcessor/src/main/java/com/group7/FileProcessor/entries/RemFile.java [hunk 2]
```diff
@@ -4,40 +4,86 @@
...
-    String repositoryId;
-    static int downloadnum = 0;
+public class RemFile extends LocFile{
+    
+    private int entryId;
+    private String repositoryId;
+    private static String servicePrincipalKey = "z6N-YywpamJ5Dr227qKb";
+    private static String accessKeyBase64 = "ewoJImN1c3RvbWVySWQiOiAiMTQwMTM1OTIzOCIsCgkiY2xpZW50SWQiOiAiY2M4NWRlZGUtMGIzZS00ZGIyLTkyNzYtOWY5N2EzMGY2ZWZlIiwKCSJkb21haW4iOiAibGFzZXJmaWNoZS5jYSIsCgkiandrIjogewoJCSJrdHkiOiAiRUMiLAoJCSJjcnYiOiAiUC0yNTYiLAoJCSJ1c2UiOiAic2lnIiwKCQkia2lkIjogInh0UFBLcl8zbGVSa1dFQi1QaU9yNkFDMzJETjRRdDd0VmI4amE4VHNwRTQiLAoJCSJ4IjogImtsZl9IOWgxMTN6dmRHcjZ0Rk5MRC1wYk15YWJWUjZzTk5Fb2l1d2NmSTgiLAoJCSJ5IjogInVWcUR4NlRSV0JBaWZTbXJ6NG9uX2NXeEZlTDVPc29xM2V6U0ZQbWtkRk0iLAoJCSJkIjogInJfMHZXYnB6Q2Z2YUYzYXZQeERhNzdKTzItUkRrSWctRnpHWENrbFRNQzgiLAoJCSJpYXQiOiAxNjc3Mjk3NTA4Cgl9Cn0=";
+    private static AccessKey accessKey = AccessKey.createFromBase64EncodedAccessKey(accessKeyBase64);
+    private static int downloadnum=0;
```
Comment (Tacitor) [inline FileProcessor/src/main/java/com/group7/FileProcessor/entries/RemFile.java:32]: please remove this

File: FileProcessor/src/main/java/com/group7/FileProcessor/entries/testmain.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+/**
+ *
+ * @author shale
+ */
+public class testmain {
+    public static void main(String[] args) throws IOException{
+        RemFile file = new RemFile(25,"r-0001d410ba56");
+        System.out.println(file.contents);
+    }
+}
```
Comment (Tacitor) [inline FileProcessor/src/main/java/com/group7/FileProcessor/entries/testmain.java:18]: At some point this class should get deleted

File: FileProcessor/src/main/java/com/group7/FileProcessor/entries/RemFolder.java [hunk 4]
```diff
@@ -5,35 +5,160 @@
...
  * @author shale
  */
 public class RemFolder extends LocFolder {
-
-    String accessKey;
+    
+    int entryId;
     String repositoryId;
-    static int downloadnum = 0;
+    ArrayList<Integer> entryIds = new ArrayList<Integer>();
```
Comment (Tacitor) [inline FileProcessor/src/main/java/com/group7/FileProcessor/entries/RemFolder.java:29]: Is this a list of just the child folders? @Shalev001

**Pull request merged**